api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -486,6 +489,52 @@ public void setDiagnoses(Set<Diagnosis> diagnoses) {
 		this.diagnoses = diagnoses;
 	}
 	
+	/**
+	 * Basic property getter for conditions
+	 * 
+	 * @return conditions - set of conditions
+	 * @since 2.4.0, 2.3.1
+	 */
+	public Set<Condition> getConditions() {
+		if (conditions == null) {
+			conditions = new LinkedHashSet<>();
+		}
+		return conditions;
+	}
+
+	/**
+	 * Basic property setter for conditions
+	 *  
+	 * @param conditions - set of conditions
+	 * @since 2.4.0, 2.3.1
+	 */
+	public void setConditions(Set<Condition> conditions) {
+		this.conditions = conditions;
+	}
+
+	/**
+	 * Add the given Condition to the set of conditions for this Encounter
+	 *
+	 * @param condition - the condition to add
+	 */
+	public void addCondition(Condition condition) {
+		if (condition != null) {
```

Comment:
I would not silently ignore one passing a null parameter. In other words, i would remove the if check and assume that the condition is not null. If it turns out to be null, better for the caller to automatically get a NullPointerException than chewing it quietly.